### ISSUE TITLE
`MisleadingEscapedSpace` producing false positives

### ISSUE THREAD

**Issue by PBoddington** [google/error-prone#4631]
With the introduction of the new rule `MisleadingEscapedSpace` in errorprone version `2.34.0`, we're finding that code such as the following is considered an error.

```
static final String MY_STRING = """
    ONE
    TWO
    THREE\s""";
```

The error is 
```
error: [MisleadingEscapedSpace] Using \s anywhere except at the end of a line in a text block is potentially misleading.
```

However, if you use an ordinary space character instead of `\s`, the character is stripped, so this seems to be the intended way to use `\s` in a text block.

This doesn't happen with the code 

```
static final String MY_STRING = """
    ONE
    TWO
    THREE\s
    """;
```

so it appears to be an issue only affecting the case that the final `"""` is not on its own line.

**Comment by cushon** [comment]
cc @graememorgan

**Comment by graememorgan** [comment]
Thanks for the report! I'll investigate and send a fix.

**Comment by graememorgan** [comment]
Fixed at head. :)
